File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/service/JsonConverterService.java
package com.raxim.myscoutee.profile.service;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

public class JsonConverterService {

    private final ObjectMapper objectMapper;

    public JsonConverterService(ObjectMapper objectMapper) {
        this.objectMapper = objectMapper;
    }

    public <T> T jsonToObject(String jsonString, Class<T> targetClass) {
        try {
            return objectMapper.readValue(jsonString, targetClass);
        } catch (IOException e) {
            // Handle the exception appropriately
            e.printStackTrace();
            return null;
        }
    }
}

File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/service/JsonConverterService.kt
package com.raxim.myscoutee.profile.service

import com.fasterxml.jackson.databind.ObjectMapper
import java.io.IOException

class JsonConverterService(private val objectMapper: ObjectMapper) {
    fun <T> jsonToObject(jsonString: String?, targetClass: Class<T>?): T? {
        return try {
            objectMapper.readValue<T>(jsonString, targetClass)
        } catch (e: IOException) {
            // Handle the exception appropriately
            e.printStackTrace()
            null
        }
    }
}